lib/generators/cable_ready/channel_generator.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+class CableReady::ChannelGenerator < Rails::Generators::NamedBase
+  source_root File.expand_path("templates", __dir__)
+
+  class_option :stimulus, type: :boolean
+  class_option :broadcast_to, type: :boolean
+  class_option :resource, type: :string
+  class_option :identifier, type: :string
+
+  def create_channel
+    generate "channel", file_name
+
+    if using_broadcast_to?
+      resource = ask_for_resource
+      resource_name = resource.underscore
+
+      gsub_file "app/channels/#{file_name}_channel.rb", %r{# stream_from.*\n}, "stream_for #{resource}.find(params[:id])\n"
+
+      if using_stimulus?
+        template "app/javascript/controllers/%file_name%_controller.js"
+      end
+    else
+      identifier = ask_for_identifier
+
+      prepend_to_file "app/javascript/channels/#{file_name}_channel.js", "import CableReady from 'cable_ready'\n"
+      inject_into_file "app/javascript/channels/#{file_name}_channel.js", after: "// Called when there's incoming data on the websocket for this channel\n" do
+        <<-JS
+    if (data.cableReady) CableReady.perform(data.operations)
+        JS
+      end
+
+      gsub_file "app/channels/#{file_name}_channel.rb", %r{# stream_from.*\n}, "stream_from \"#{identifier}\"\n"
+    end
+  end
+
+  private
+
+  def using_broadcast_to?
+    @using_broadcast_to ||= options.fetch(:broadcast_to) {
```

Comment:
Should we explain the use of `options` in the `USAGE` file?

Reply:
The options get generated by rails itself. Unless you mean to explicitly add some examples?

Reply:
So it I run —help, it will show me those options? Sorry, haven’t had the time to check it

Reply:
Yes!

```bash
❯ rails g cable_ready:channel                                                                                                                                                 15:42:34
Usage:
  rails generate cable_ready:channel NAME [options]

Options:
  [--skip-namespace], [--no-skip-namespace]  # Skip namespace (affects only isolated applications)
  [--stimulus], [--no-stimulus]              # Indicates when to generate stimulus
  [--broadcast-to], [--no-broadcast-to]      # Indicates when to generate broadcast to
  [--resource=RESOURCE]                      # Indicates when to generate resource
  [--identifier=IDENTIFIER]                  # Indicates when to generate identifier

Runtime options:
  -f, [--force]                    # Overwrite files that already exist
  -p, [--pretend], [--no-pretend]  # Run but do not make any changes
  -q, [--quiet], [--no-quiet]      # Suppress status output
  -s, [--skip], [--no-skip]        # Skip files that already exist

Description:
  Create an ActionCable channel ready for usage with CableReady

Example:
  bin/rails generate cable_ready:channel Thing

  This will create:
    app/channels/thing_channel.rb
    app/javascript/channels/thing_channel.js
    app/javascript/controllers/thing_controller.js (Optional)
```